src/components/MovieGallery/MovieGallery.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useContext, useEffect, useState } from 'react'
+import style from './Moviegallery.module.scss'
+import MovieCard from '../MovieCard/MovieCard'
+import { ContextMain } from '../Context/ContextMain'
+import { ContextData, DataProvider } from '../Context/ContextData'
+
+
+function MovieGallery() {
+  const valueData = useContext(ContextData)
+  const value = useContext(ContextMain)
+  let movieItems = valueData.movieData
+
```

Comment:
you need to use state to hold movies and filter them by user's action. Like this you are not using React's capabilities. https://beta.reactjs.org/learn/state-a-components-memory